### ISSUE TITLE
Build azure iot sdk in Yocto

### ISSUE THREAD

**Issue by bcis93** [atharvanan1/aesd_project_build#10]
We need to build the [azure iot c sdk](https://github.com/Azure/azure-iot-sdk-c) in Yocto so that we can use the resulting libraries to build our [azure app](https://github.com/atharvanan1/aesd_project_build/issues/6). We either need to build this from the source code or find an existing Yocto layer that we can bring in

DoD: Azure iot sdk added to Yocto build. Builds successfully

**Comment by bcis93** [comment]
There is an existing yocto layer from intel ([meta-iot-cloud](https://github.com/intel-iot-devkit/meta-iot-cloud)) that should be able to build the azure iot sdk. This layer contains a lot of unnecessary things (like support for AWS, google cloud, etc), but we can just include the relevant packages/recipes into our build

**Comment by bcis93** [comment]
There are only two packages that we actually need from this layer: azure-iot-sdk-c and azure-iot-sdk-c-dev

**Comment by bcis93** [comment]
I ran a build using the [thud branch](https://github.com/intel-iot-devkit/meta-iot-cloud/tree/thud) because that is what we are building everything with, and it built, but I found that it built a very outdated version of the sdk. It looks like the warrior branch builds a more up-to-date version of the sdk, so I will see if that branch is compatible with our build

**Comment by bcis93** [comment]
Warrior branch also builds in Yocto! Added to build with a6818d97671e6fdfd3416b69f86a1cbbbea091fe.

Note that we still need to make sure that our azure application is compatible with the version of azure-iot-sdk-c that this layer builds, but that will be handled in [this issue](https://github.com/atharvanan1/aesd_project_build/issues/6). For now, we have found a way to build the sdk in Yocto, so I am closing this issue
